From akpm@osdl.org  Tue Dec 19 22:27:17 2006
Message-Id: <200612192112.kBJLC8VY010608@shell0.pdx.osdl.net>
From: Adrian Bunk <bunk@stusta.de>
Subject: PCI: don't export device IDs to userspace
To: greg@kroah.com
Cc: akpm@osdl.org, bunk@stusta.de, gregkh@suse.de
Date: Tue, 19 Dec 2006 13:12:07 -0800

From: Adrian Bunk <bunk@stusta.de>

I don't see any good reason for exporting device IDs to userspace.

Signed-off-by: Adrian Bunk <bunk@stusta.de>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/Kbuild |    2 --
 include/linux/pci.h  |    6 +++---
 2 files changed, 3 insertions(+), 5 deletions(-)

--- gregkh-2.6.orig/include/linux/Kbuild
+++ gregkh-2.6/include/linux/Kbuild
@@ -119,7 +119,6 @@ header-y += nfs4_mount.h
 header-y += nfs_mount.h
 header-y += oom.h
 header-y += param.h
-header-y += pci_ids.h
 header-y += pci_regs.h
 header-y += personality.h
 header-y += pfkeyv2.h
@@ -162,7 +161,6 @@ header-y += vt.h
 header-y += wireless.h
 header-y += xattr.h
 header-y += x25.h
-header-y += zorro_ids.h
 
 unifdef-y += acct.h
 unifdef-y += adb.h
--- gregkh-2.6.orig/include/linux/pci.h
+++ gregkh-2.6/include/linux/pci.h
@@ -20,9 +20,6 @@
 /* Include the pci register defines */
 #include <linux/pci_regs.h>
 
-/* Include the ID list */
-#include <linux/pci_ids.h>
-
 /*
  * The PCI interface treats multi-function devices as independent
  * devices.  The slot/function address of each device is encoded
@@ -54,6 +51,9 @@
 #include <asm/atomic.h>
 #include <linux/device.h>
 
+/* Include the ID list */
+#include <linux/pci_ids.h>
+
 /* File state for mmap()s on /proc/bus/pci/X/Y */
 enum pci_mmap_state {
 	pci_mmap_io,
